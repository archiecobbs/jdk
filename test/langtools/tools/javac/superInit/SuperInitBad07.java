/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Test invalid reference to "this" prior to superclass construction
 *
 * @compile/fail/ref=SuperInitBad07.out -XDrawDiagnostics SuperInitBad07.java
 */
class SuperInitBad07 {

    SuperInitBad07() {
        System.identityHashCode(this);
        super();
    }
}
