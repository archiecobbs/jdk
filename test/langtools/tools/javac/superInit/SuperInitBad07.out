SuperInitBad07.java:11:33: compiler.err.cant.ref.before.ctor.called: this
1 error
